nixos/modules/system/etc/etc.nix

```diff
@@ -8,21 +8,43 @@ let
 
   etc' = filter (f: f.enable) (attrValues config.environment.etc);
 
-  etc = pkgs.stdenvNoCC.mkDerivation {
-    name = "etc";
-
-    builder = ./make-etc.sh;
-
+  etcEntryBuilder = { target, source, mode, user, group, ... }:
+    if hasInfix "*" source then
+      # If the source name contains '*', perform globbing.
+      /* sh */ ''
+        mkdir -p "$out/etc/${target}"
```

Comment:
Should the arguments be escaped here?

Reply:
Why not? I really don't want `target` to be globbed. It's `source` that's supposed to be globbed.

Reply:
Just to be sure we are on the same page. Right now `target` is quoted however not escaped. It could be escaped as above.

Reply:
We were not on the same page, I fixed that now